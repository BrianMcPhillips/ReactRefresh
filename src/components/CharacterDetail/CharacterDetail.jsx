import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ name, image }) => (
  <>
    <img src={image} alt={name} />
    <h2>{name}</h2>
  </>
);

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default CharacterDetail;
