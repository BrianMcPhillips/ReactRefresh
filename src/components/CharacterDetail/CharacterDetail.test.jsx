import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  afterEach(() => cleanup());
  it('renders CharacterDetail', () => {
    const { asFragment } = render(<CharacterDetail 
      name="Test"
      image="http://placecage.com/200/200"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
