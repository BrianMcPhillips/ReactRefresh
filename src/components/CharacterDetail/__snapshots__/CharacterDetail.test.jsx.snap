// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <img
    alt="Test"
    src="http://placecage.com/200/200"
  />
  <h2>
    Test
  </h2>
</DocumentFragment>
`;
